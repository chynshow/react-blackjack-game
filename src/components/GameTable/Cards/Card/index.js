import React from 'react';

const Card = ({ card, cardImgStyles }) => {
  return (
    <div className='c-card'>
      <img src={card?.image} alt={card?.code} className={cardImgStyles} />
    </div>
  );
};

export default React.memo(Card);
