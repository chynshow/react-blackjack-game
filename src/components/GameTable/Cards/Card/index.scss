.c-card {
  animation: scale 1000ms linear both;
  &__img {
    margin: 0.5rem;
    width: 5rem;
  }
}
