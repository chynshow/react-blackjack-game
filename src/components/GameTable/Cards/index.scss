.c-cards {
  padding: 1rem;
  display: flex;
  flex-direction: column;
  align-items: center;

  &__owner-name {
    padding: 0.5rem;
    font-weight: 400;
    display: flex;
    align-items: center;
  }

  &__owner-score {
    margin-left: 0.5rem;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 2.5rem;
    height: 2.5rem;
    border-radius: 50%;
    background: $text-color;
    color: $bg-color;
  }
  &__container {
    display: flex;
  }

  &--disable {
    filter: grayscale(100%);
  }
}

.c-card {
  &__img {
    // filter: grayscale(100%);
    margin: 0.5rem;
    width: 5rem;
  }
}
