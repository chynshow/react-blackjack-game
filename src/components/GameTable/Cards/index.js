import React from 'react';
import Card from './Card/index';

const Cards = ({ cards, ownerName, ownerScore, className }) => {
  return (
    <div className={className ? className : 'c-cards'}>
      <h3 className='c-cards__owner-name'>
        {ownerName} <span className='c-cards__owner-score'>{ownerScore}</span>
      </h3>

      <div className='c-cards__container'>
        {cards.map((card) => (
          <Card key={card.code} card={card} cardImgStyles='c-card__img' />
        ))}
      </div>
    </div>
  );
};

export default React.memo(Cards);
