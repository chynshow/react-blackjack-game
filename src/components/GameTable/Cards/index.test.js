import React from 'react';
import Cards from './index';
import { shallow } from 'enzyme';

it('render Cards component', () => {
  const c = shallow(<Cards />);
  expect(c).toMatchSnapshot();
});
