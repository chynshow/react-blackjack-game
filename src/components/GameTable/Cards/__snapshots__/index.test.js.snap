// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Cards component 1`] = `
<div
  className="c-cards"
>
  <h3
    className="c-cards__owner-name"
  >
     
    <span
      className="c-cards__owner-score"
    />
  </h3>
  <div
    className="c-cards__container"
  />
</div>
`;
